dev/PackageManager/API/PackageManager.idl

```diff
@@ -0,0 +1,383 @@
+
+    // Requires Windows >- 10.0.19041.0 (aka 2004 aka 20H1)
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass RegisterPackageOptions
+    {
+        RegisterPackageOptions();
+
+        PackageVolume AppDataVolume;
+        IVector<Windows.Foundation.Uri> DependencyPackageUris { get; };
+        IVector<String> OptionalPackageFamilyNames { get; };
+        Windows.Foundation.Uri ExternalLocationUri;
+        Boolean DeveloperMode;
+        Boolean ForceAppShutdown;
+        Boolean ForceTargetAppShutdown;
+        Boolean ForceUpdateFromAnyVersion;
+        Boolean InstallAllResources;
+        Boolean StageInPlace;
+        Boolean AllowUnsigned;
+        Boolean DeferRegistrationWhenPackagesAreInUse;
+
+        Boolean IsExpectedDigestsSupported { get; };            // Requires Windows >= 10.0.22621.0 (aka Win11 22H2)
+        IMap<Windows.Foundation.Uri, String> ExpectedDigests{ get; };
+    }
+
+    // Requires Windows >- 10.0.19041.0 (aka 2004 aka 20H1)
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass RemovePackageOptions
+    {
+        RemovePackageOptions();
+
+        Boolean PreserveApplicationData;
+        Boolean PreserveRoamableApplicationData;
+        Boolean RemoveForAllUsers;
+        DeploymentPriority Priority;
+    }
+
+    // The API surface is strongly consistent:
+    // 1. Modification operations involving incoming packages...
+    //      - <verb>PackageAsync       (string, options)
+    //      - <verb>PackageByUriAsync  (uri, options)
+    //      - <verb>PackageSetAsync    (packageset, options)
+    //      - <verb>PackageSetByIdAsync(id, options)
+    //    where
+    //      <verb> = Add | Register | Stage
+    // 2. Modification operations involving existing packages...
+    //      - <verb>PackageAsync                   (string, options)
+    //      - <verb>PackageByPackageFamilyNameAsync(string, options)
+    //      - <verb>PackageByPackageFullNameAsync  (string, options)
+    //    where
+    //      <verb> = Deprovision | Provision | Register | Remove
+    //
+    // TODO: AppInstaller
+    // TODO: Move
+    // TODO: Find
+    // TODO: Properties (Status, StubPreference)
+    // TODO: RequestAdd
+    //
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageDeploymentManager
+    {
+        // Get an instance of the manager
+        static PackageDeploymentManager GetDefault();
+
+        // Return true if the package set's content are present and available for use
+        Boolean IsPackageSetReady(PackageSet packageSet);
+
+        Boolean IsPackageSetReadyById(String packageSetId);
+
+        // Check if the necessary package(s) needed by the scenario are present
+        // and available for use and if not then Make It So.
+        // If the necessary packages(s) are not present on the system
+        // then make them available (download, install, etc).
+        // If the necessary AI Models are present and available this is equivalent to IsReady(id).
```

Comment:
Is this limited to AI, or would any package set be available here?

Reply:
None of this should be limited to a specific type of packageset.

Reply:
That's a typo from an early doc I started from. Carl's right, this API is about MSIX packages regardless of their content or use. Fixed the comment.